hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/CommitsCommand.java

```diff
@@ -206,23 +207,29 @@ public boolean isCompareCommitsAvailable() {
     public String compareCommits(
         @CliOption(key = {"path"}, help = "Path of the dataset to compare to")
         final String path) throws Exception {
-        HoodieTableMetadata target = new HoodieTableMetadata(HoodieCLI.fs, path);
-        HoodieTableMetadata source = HoodieCLI.tableMetadata;
+        HoodieTableMetaClient target = new HoodieTableMetaClient(HoodieCLI.fs, path);
+        HoodieTimeline targetTimeline = target.getActiveCommitTimeline();
+        HoodieTableMetaClient source = HoodieCLI.tableMetadata;
+        HoodieTimeline sourceTimeline = source.getActiveCommitTimeline();
         String targetLatestCommit =
-            target.isCommitsEmpty() ? "0" : target.getAllCommits().lastCommit();
+            targetTimeline.getInstants().iterator().hasNext() ? "0" : targetTimeline.lastInstant().get();
         String sourceLatestCommit =
-            source.isCommitsEmpty() ? "0" : source.getAllCommits().lastCommit();
+            sourceTimeline.getInstants().iterator().hasNext() ? "0" : sourceTimeline.lastInstant().get();
 
-        if (sourceLatestCommit != null && HoodieCommits
-            .isCommit1After(targetLatestCommit, sourceLatestCommit)) {
+        if (sourceLatestCommit != null && sourceTimeline
+            .compareInstants(targetLatestCommit, sourceLatestCommit, HoodieTimeline.GREATER)) {
             // source is behind the target
-            List<String> commitsToCatchup = target.findCommitsSinceTs(sourceLatestCommit);
-            return "Source " + source.getTableName() + " is behind by " + commitsToCatchup.size()
+            List<String> commitsToCatchup =
```

Comment:
Commit will be a special type of Instant. I think this becomes clear with the refactor. Take a look at it again and let me know what you think.